src/lib/arch/aarch64/InstructionMetadata.cc

```diff
@@ -362,10 +362,12 @@ InstructionMetadata::InstructionMetadata(const cs_insn& insn)
       break;
     case Opcode::AArch64_RET:
       // RET doesn't list use of x30 (LR) if no register is supplied
-      operandCount = 1;
-      operands[0].type = ARM64_OP_REG;
-      operands[0].reg = ARM64_REG_LR;
-      operands[0].access = CS_AC_READ;
+      if (operandCount == 0) {
+        operandCount = 1;
+        operands[0].type = ARM64_OP_REG;
+        operands[0].reg = ARM64_REG_LR;
+        operands[0].access = CS_AC_READ;
+      }
```

Comment:
Should probably add to the tests (or create some if none are present) to check if doing `ret xm` works correctly.